app/grandchallenge/challenges/forms.py

```diff
@@ -534,3 +535,26 @@ def __init__(self, *args, **kwargs):
             self.fields["status"].widget.attrs["disabled"] = True
         self.helper = FormHelper(self)
         self.helper.layout.append(Submit("save", "Save"))
+
+    def clean_status(self):
+        status = self.cleaned_data.get("status")
+        if (
+            self.instance._orig_status
+            == self.instance.ChallengeRequestStatusChoices.PENDING
+            and self.instance._orig_status != status
+        ):
+            if status == self.instance.ChallengeRequestStatusChoices.ACCEPTED:
+                try:
+                    challenge = self.instance.create_challenge()
+                except ValidationError:
+                    raise ValidationError(
```

Comment:
Why catch and raise the same thing? The underlying Validation method should give the correct error. Maybe you need to clean the short name instead? Or is that already done?

Reply:
The short name is cleaned, but it only checks if a challenge request with the same short name exists, not if a challenge with the same short name exists. I could add that to the validation, but I think I should wait with doing that until we stop asking organizers of already existing challenges to file a request in retrospect. They would then have to come up with a new short name and things would get a little messy. Of course, for brand new challenges is makes sense to also check whether the short name exists among challenges.

Reply:
And as per raising the ValidationError, I'm doing this to add a custom error message (adding the contact support part).